homeassistant/components/govee_light_api/light.py

```diff
@@ -0,0 +1,173 @@
+"""Govee Light."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from typing import Any
+
+from govee_local_api import GoveeDevice
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_COLOR_TEMP_KELVIN,
+    ATTR_RGB_COLOR,
+    ColorMode,
+    LightEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC, DeviceInfo
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .capability import (
+    GOVEE_COORDINATORS_MAPPER,
+    GOVEE_DEVICE_CAPABILITIES,
+    GoveeLightCapabilities,
+)
+from .const import DISPATCH_GOVEE_LIGHT_DISCOVERED, DOMAIN, MANUFACTURER
+from .coordinator import GoveeLocalApiCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Govee light setup."""
+
+    coordinator: GoveeLocalApiCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_entities(
+        GoveeLight(coordinator, device) for device in coordinator.devices
+    )
+
+    def discovery_callback(device: GoveeDevice, is_new: bool) -> bool:
+        if is_new:
+            entity = GoveeLight(coordinator, device)
+            async_add_entities([entity])
+            async_dispatcher_send(hass, DISPATCH_GOVEE_LIGHT_DISCOVERED, entity)
```

Comment:
Why do we dispatch this?

Reply:
I took it from the other integration, but if it's not required or needed I'll remove it

Reply:
Oh, I take another look at the code and I totally misunderstood the use of `async_dispatcher_send`. I'll remove it